### PR TITLE
[Salsa] Don't put things in CommonJS modules into the global scope

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
/cc @mhegazy @vladima

### CODE REVIEW

File: src/compiler/binder.ts [hunk 1]
```diff
@@ -210,6 +216,24 @@ namespace ts {
...
         }
 
+        /**
+         * Returns the first invocation of an identifier 'require' with one argument,
+         * or undefined if that none exist.
+        */
+        function findCommonJSRequireCalls(node: Node): Node {
+            if (node.kind === SyntaxKind.CallExpression) {
+                let call = <CallExpression>node;
+                if(call.expression.kind === SyntaxKind.Identifier &&
```
Comment (DanielRosenwasser) [inline src/compiler/binder.ts]: Space after `if`

File: src/compiler/binder.ts [hunk 2]
```diff
@@ -210,6 +216,24 @@ namespace ts {
...
+                    (<Identifier>call.expression).text === 'require' &&
+                    call.arguments.length === 1) {
```
Comment (DanielRosenwasser) [inline src/compiler/binder.ts]: I think you also want to check whether the argument is a string literal.

Comment (DanielRosenwasser) [inline src/compiler/binder.ts]: Actually, never mind, since this is just for ensuring module scoping rules, right?

Comment (RyanCavanaugh) [inline src/compiler/binder.ts]: Correct. I'm ambivalent on checking the argument syntactic form

**Comment by DanielRosenwasser** [comment]
:+1:

File: src/compiler/binder.ts [hunk 3]
```diff
@@ -210,6 +216,24 @@ namespace ts {
             return propAccess.name.text;
         }
 
+        /**
+         * Returns the first invocation of an identifier 'require' with one argument,
+         * or undefined if that none exist.
+        */
```
Comment (yuit) [inline src/compiler/binder.ts]: indent

File: src/compiler/binder.ts [hunk 4]
```diff
@@ -210,6 +216,24 @@ namespace ts {
...
+                let call = <CallExpression>node;
+                if(call.expression.kind === SyntaxKind.Identifier &&
+                    (<Identifier>call.expression).text === 'require' &&
+                    call.arguments.length === 1) {
+
+                    return node;
+                }
+            }
+
+            return forEachChild(node, findCommonJSRequireCalls);
```
Comment (zhengbli) [inline src/compiler/binder.ts]: Can you give an example of the `require` appearing at the child position? Do we need a test for that case?

**Comment by vladima** [comment]
:+1:

File: src/compiler/binder.ts [hunk 5]
```diff
@@ -210,6 +216,24 @@ namespace ts {
             return propAccess.name.text;
         }
 
+        /**
+         * Returns the first invocation of an identifier 'require' with one argument,
+         * or undefined if that none exist.
+         */
+        function findCommonJSRequireCalls(node: Node): Node {
```
Comment (mhegazy) [inline src/compiler/binder.ts]: so why not put this in the parser, so that SourceFile.externalModuleIndicator is sufficient all the time?

**Pull request merged**